Modules/System/WebDAV/src/Model/WebDAVContentParser.Codeunit.al

```diff
@@ -0,0 +1,107 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+codeunit 5682 "WebDAV Content Parser"
+{
+
+    Access = Internal;
+
+    [NonDebuggable]
+    procedure Initialize(InitBaseUrl: Text; InitOnlyFiles: Boolean; InitOnlyCollections: Boolean)
+    begin
+        BaseUrl := InitBaseUrl;
+        OnlyFiles := InitOnlyFiles;
+        OnlyCollections := InitOnlyCollections;
+    end;
+
+    [NonDebuggable]
+    procedure Parse(Payload: Text; var WebDAVContent: Record "WebDAV Content")
+    var
+        XmlDoc: XmlDocument;
+        ChildNodes: XmlNodeList;
+        ResponseNode: XmlNode;
+    begin
+        XmlNamespaceManager.AddNamespace('dav', 'DAV:');
+
+        XmlDocument.ReadFrom(Payload, XmlDoc);
+        XmlDoc.SelectNodes('//dav:response', XmlNamespaceManager, ChildNodes);
+        foreach ResponseNode in ChildNodes do
+            ParseSingle(ResponseNode, WebDAVContent);
+    end;
+
+    [NonDebuggable]
+    procedure ParseSingle(XmlNode: XmlNode; var WebDAVContent: Record "WebDAV Content")
+    begin
+        WebDAVContent.Init();
+        WebDAVContent."Is Collection" := HasNode(XmlNode, 'dav:propstat/dav:prop/dav:resourcetype/dav:collection');
+
+        if OnlyCollections then
+            if not WebDAVContent."Is Collection" then
+                exit;
```

Comment:
I think it makes sense to return whether the single was parsed or not. In this case exit(false)